addons/hr_gamification/models/hr_employee.py

```diff
@@ -0,0 +1,44 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from openerp import api, fields, models
+
+
+class HrEmployee(models.Model):
+    _name = "hr.employee"
+    _inherit = "hr.employee"
+
+    goal_ids = fields.One2many('gamification.goal', string='Employee HR Goals', compute='_compute_employee_goals')
+    badge_ids = fields.One2many('gamification.badge.user', string='Employee Badges', compute='_compute_employee_badges')
+    has_badges = fields.Boolean(compute='_compute_has_badges')
+
+    @api.multi
+    def _compute_employee_goals(self):
+        for goal in self:
+            goal_ids = self.env['gamification.goal'].read_group([
+                ('user_id', 'in', goal.user_id.ids),
+                ('challenge_id.category', '=', 'hr')
+            ], [], [])
+            data = dict([('goal_id', goal_id['id']) for goal_id in goal_ids])
+            goal.goal_ids = [(6, 0,  [data['goal_id']])]
+
+    @api.multi
+    def _compute_employee_badges(self):
+        for badge in self:
+            badge.badge_ids = self.env['gamification.badge.user'].search([
+                '|',
+                ('employee_id', '=', badge.id),
+                '&',
+                ('employee_id', '=', False),
+                ('user_id', '=', badge.user_id.id)
+            ])
+
+    @api.multi
+    def _compute_has_badges(self):
+        for employee in self:
+            employee_badges_count = self.env['gamification.badge.user'].read_group([
```

Comment:
same here. read_group inside loop bad idea. even you used  "('employee_id', 'in', employee.ids)".